src/modules/map/services/OlMeasurementHandler.js

```diff
@@ -0,0 +1,203 @@
+import Draw from 'ol/interaction/Draw';
+import { Vector as VectorSource } from 'ol/source';
+import { Vector as VectorLayer } from 'ol/layer';
+import { Fill, Stroke, Style } from 'ol/style';
+import { unByKey } from 'ol/Observable';
+import { LineString, Polygon, Circle, LinearRing } from 'ol/geom';
+import Overlay from 'ol/Overlay';
+import { getLength } from 'ol/sphere';
+
+
+const ZPOLYGON = 10;
+const ZLINE = 20;
+
+//todo: find a better place....move to utils
+export const getGeometryLength = (geometry) => {
+	let lineString;
+	if(geometry instanceof LineString) {
+		lineString = geometry;
+	}
+	else if(geometry instanceof LinearRing) {
+		lineString = new LineString(geometry.getCoordinates());
+	}
+	else if(geometry instanceof Polygon) {
+		lineString = new LineString(geometry.getLinearRing(0).getCoordinates());
+	}	
+	
+	if(lineString) {
+		return lineString.getLength();
+	}
+	return 0;
+};
+
+
+//todo: find a better place....move to utils
+export const canShowAzimuthCircle = (geometry) => {
+	if(geometry instanceof LineString) {
+		const coords = geometry.getCoordinates();
+		if(coords.length === 2 || 
+			(coords.length === 3 && coords[1][0] === coords[2][0] && coords[1][1] === coords[2][1])) {
+			return true;
+		}
+	}
+	return false;
+};
+
+//todo: find a better place....maybe StyleService
+export const measureStyleFunction = (feature) => {
+	
+	const color = [255, 0, 0];
+	const stroke = new Stroke({
+		color:color.concat([1]),
+		width:1
+	});
+
+	const dashedStroke = new Stroke({
+		color:color.concat([1]),
+		width:3,
+		lineDash:[8]
+	});
+	
+	const zIndex = (feature.getGeometry() instanceof LineString) ?	ZLINE : ZPOLYGON;
+
+	const styles = [
+		new Style({
+			fill: new Fill({ 
+				color:color.concat([0.4]) 
+			}),
+			stroke:dashedStroke,
+			zIndex:zIndex
+		}),
+		new Style({
+			stroke:stroke,
+			geometry: feature => {
+				
+				if(canShowAzimuthCircle(feature.getGeometry())) {					
+					const coords = feature.getGeometry().getCoordinates();
+					const radius = getGeometryLength(feature.getGeometry());
+					const circle = new Circle(coords[0], radius);
+					return circle;
+				}
+			},
+			zIndex:0
+		})];
+
+	return styles;
+};
+export class OlMeasurementHandler {
+	//this handler could be statefull
+	constructor() {
+		this._vectorLayer = null;
+		this._draw = false;				
+		this._measureTooltip;		
+	}
+
+	/**
+	 * Activates the Handler.
+	 * @param {Map} olMap 
+	 * @returns {VectorLayer} the olLayer which should be attached to
+	 */
+	// eslint-disable-next-line no-unused-vars
+	activate(olMap) {
+		const prepareInteraction = (map) => {
+			const source = new VectorSource({ wrapX: false });	
+			const layer = new VectorLayer({
+				source: source,
+				style: measureStyleFunction,
+				map:map // Sets the layer as overlay on a map
```

Comment:
Please remove adding the layer to the map.
With the latest commits, the layer is added by the layer managment in the background

Reply:
if this line is removed, the drawed line will be disappear. The feature is in the layer, but is not rendered.